src/solvers/flattening/boolbv_complex.cpp

```diff
@@ -6,11 +6,14 @@ Author: Daniel Kroening, kroening@kroening.com
 
 \*******************************************************************/
 
-
 #include "boolbv.h"
 
+#include <util/invariant.h>
+
 bvt boolbvt::convert_complex(const exprt &expr)
 {
+  PRECONDITION(expr.id() == ID_complex);
```

Comment:
The method should take a `complex_exprt` as parameter, making various checks unnecessary.